src/v/cluster/archival/segment_reupload.cc

```diff
@@ -331,14 +331,11 @@ void segment_collector::do_collect(segment_collector_mode mode) {
         _can_replace_manifest_segment = true;
     }
 
```

Comment:
> archival: fix "Segment does not contain searched for offset" error

would you mind provideing more information in the commit message about this fix. i don't know what that error is (i don't see it in the diff below), how/when it occurs, what we are "fixing" / what was broken with the error, or how the fix works.